x-pack/plugins/infra/public/pages/metrics/inventory_view/components/node_details/tabs/logs.tsx

```diff
@@ -4,14 +4,86 @@
  * you may not use this file except in compliance with the Elastic License.
  */
 
-import React from 'react';
+import React, { useCallback, useMemo, useState } from 'react';
+import { FormattedMessage } from '@kbn/i18n/react';
 import { i18n } from '@kbn/i18n';
+import { EuiFieldSearch } from '@elastic/eui';
+import { EuiFlexGroup } from '@elastic/eui';
+import { EuiFlexItem } from '@elastic/eui';
+import { EuiButtonEmpty } from '@elastic/eui';
 import { TabContent, TabProps } from './shared';
+import { LogStream } from '../../../../../../components/log_stream';
+import { useWaffleOptionsContext } from '../../../hooks/use_waffle_options';
+import { findInventoryFields } from '../../../../../../../common/inventory_models';
+import { euiStyled } from '../../../../../../../../observability/public';
+import { useLinkProps } from '../../../../../../hooks/use_link_props';
+import { getNodeLogsUrl } from '../../../../../link_to';
 
 const TabComponent = (props: TabProps) => {
-  return <TabContent>Logs Placeholder</TabContent>;
+  const [textQuery, setTextQuery] = useState('');
+  const endTimestamp = props.currentTime;
+  const startTimestamp = endTimestamp - 60 * 60 * 1000; // 15 minutes
+  const { nodeType } = useWaffleOptionsContext();
+  const { options, node } = props;
+
+  const filter = useMemo(() => {
+    let query = options.fields
+      ? `${findInventoryFields(nodeType, options.fields).id}: "${node.id}"`
+      : ``;
+
+    if (textQuery) {
+      query += ` and message: ${textQuery}`;
+    }
+    return query;
+  }, [options, nodeType, node.id, textQuery]);
+
+  const onQueryChange = useCallback((e: React.ChangeEvent<HTMLInputElement>) => {
+    setTextQuery(e.target.value);
+  }, []);
+
+  const nodeLogsMenuItemLinkProps = useLinkProps(
+    getNodeLogsUrl({
+      nodeType,
+      nodeId: node.id,
+      time: startTimestamp,
+    })
+  );
+
+  return (
+    <TabContent>
+      <EuiFlexGroup gutterSize={'none'} alignItems="center">
+        <EuiFlexItem grow={true}>
+          <QueryWrapper>
+            <EuiFieldSearch
+              fullWidth={true}
```

Comment:
```suggestion
              fullWidth
```

Idk I just like the boolean shorthand for props